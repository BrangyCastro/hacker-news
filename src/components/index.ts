export { Header } from "./Header/Header";
export { Tabs } from "./Tabs/Tabs";
export { Panel } from "./Tabs/Panel";
export { Card } from "./Card/Card";
