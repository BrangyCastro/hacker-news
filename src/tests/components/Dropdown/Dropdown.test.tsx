import React from "react";
import { shallow } from "enzyme";
import { Dropdown } from "../../../components";
import { options } from "../../fixtures/demoOptions";

describe("Dropdown tests", () => {
  const onChange = jest.fn();

  const wrapper = shallow(
    <Dropdown
      options={options}
      placeholder="Select your news"
      onChange={onChange}
    />
  );

  test("should be displayed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should be three options", () => {
    wrapper.find("#selectField").simulate("click");
    const dropdownMenu = wrapper.find(".options");
    expect(dropdownMenu).toHaveLength(options.length);
    wrapper.find("#selectField").simulate("click");
  });

  test("should select the first option", () => {
    wrapper.find("#selectField").simulate("click");
    wrapper.find(".options").at(0).simulate("click");
    const selectValue = wrapper.find("p").text().trim();
    expect(selectValue).toBe(options[0].value);
  });
});
