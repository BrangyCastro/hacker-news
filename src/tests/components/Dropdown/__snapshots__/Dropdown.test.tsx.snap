// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown tests should be displayed correctly 1`] = `
<div
  className="selector"
>
  <div
    id="selectField"
    onClick={[Function]}
  >
    <p>
      Select your news
    </p>
    <img
      alt=""
      className=""
      src="arrow.png"
    />
  </div>
</div>
`;
